File: C++/Baek_Jun/Baek_1931.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main(int argc, char const *argv[])
{
    int n;
    scanf("%d", &n);
    vector<pair<int, int>> arr;

    for (int i = 0; i < n; i++)
    {
        int f, s;
        scanf("%d %d", &f, &s);
        arr.push_back(make_pair(f, s));
    }

    sort(arr.begin(), arr.end(), [](pair<int,int> p1, pair<int,int> p2) -> bool {
        if (p1.second < p2.second)
        {
            return true;
        }
        else if (p1.second == p2.second)
        {
            if (p1.first <= p2.first)
            {
                return true;
            }
        }
        return false;
    });

    int endTime = 0;
    int cnt = 0;
    for (int i = 0; i < n; i++)
    {
        if (arr[i].first >= endTime)
        {
            cnt++;
            endTime = arr[i].second;
        }
    }

    printf("%d", cnt);

    return 0;
}

File: Java/ProblemSolve/Baek_1931.java
import java.util.*;

public class Baek_1931 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Pair[] arr = new Pair[n];
        for (int i = 0; i < n; i++) {
            arr[i] = new Pair(sc.nextInt(), sc.nextInt());
        }

        Arrays.sort(arr, (Pair o1, Pair o2) -> {
            if (o1.second < o2.second) {
                return -1;
            } else if (o1.second == o2.second) {
                if (o1.first <= o2.first)
                    return -1;
            }
            return 1;
        });

        int endTime = 0;
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            if (arr[i].first >= endTime) {
                cnt++;
                endTime = arr[i].second;
            }
        }

        System.out.println(cnt);
    }

    static class Pair {
        int first, second;

        Pair(int first, int second) {
            this.first = first;
            this.second = second;
        }
    }
}